pnpm install --ignore-package-manager-version

### Contribution

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

As a developer who maintain dependencies for many projects (company and open source), I often switch between repositories that do have different `"packageManager"` values declared.
But now I encounter errors like `ERR_PNPM_BAD_PM_VERSION  This project is configured to use v9.0.1 of pnpm. Your current pnpm is v9.0.2` where there is even only a bugfix inside pnpm and no behavior of building up `node_modules` or `pnpm-lock.yaml` has changed.

- I do not use corepack
- I have multiple computers (Arch Linux with pacman, Mac Pro with homebrew) where the pnpm version is not always the newest or in sync with each other

### Describe the solution you'd like

`v9.0.0` of pnpm shipped with a new "feature" `pnpm will now check the package.json file for a packageManager field.`
I would like to either bypass this behavior or at least pnpm should e.g. check semver range compatibility or e.g. the `pnpm-lock.yaml`'s `lockfileVersion`.

### Describe the drawbacks of your solution

It wont be strict anymore like on v9.0.0, so I would be okay if it needs to be explicitly bypassed via CLI flag or `.npmrc` setting

### Describe alternatives you've considered

_Different ways already described above._

--- a/.changeset/yellow-dolphins-battle.md
+++ b/.changeset/yellow-dolphins-battle.md
@@ -1,0 +1,6 @@
+---
+"@pnpm/cli-utils": patch
+"pnpm": patch
+---
+
+Print a hint about the `package-manager-strict` setting, when pnpm doesn't match the version specified in the `packageManager` field in `package.json`.

--- a/cli/cli-utils/src/packageIsInstallable.ts
+++ b/cli/cli-utils/src/packageIsInstallable.ts
@@ -38,7 +38,9 @@ export function packageIsInstallable (
       if (pmVersion && pnpmVersion && pmVersion !== pnpmVersion) {
         const msg = `This project is configured to use v${pmVersion} of pnpm. Your current pnpm is v${pnpmVersion}`
         if (opts.packageManagerStrict) {
-          throw new PnpmError('BAD_PM_VERSION', msg)
+          throw new PnpmError('BAD_PM_VERSION', msg, {
+            hint: 'If you want to make this check less strict, set the package-manager-strict setting to false or the COREPACK_ENABLE_STRICT env variable to 0',
+          })
         } else {
           globalWarn(msg)
         }

--- a/cli/cli-utils/src/packageIsInstallable.ts
+++ b/cli/cli-utils/src/packageIsInstallable.ts
@@ -39,7 +39,7 @@ export function packageIsInstallable (
         const msg = `This project is configured to use v${pmVersion} of pnpm. Your current pnpm is v${pnpmVersion}`
         if (opts.packageManagerStrict) {
           throw new PnpmError('BAD_PM_VERSION', msg, {
-            hint: 'If you want to make this check less strict, set the package-manager-strict setting to false or the COREPACK_ENABLE_STRICT env variable to 0',
+            hint: 'If you want to bypass this version check, you can set the "package-manager-strict" configuration to "false" or set the "COREPACK_ENABLE_STRICT" environment variable to "0"',
           })
         } else {
           globalWarn(msg)

